scipy/meson.build

```diff
@@ -144,6 +144,28 @@ fortranobject_dep = declare_dependency(
   compile_args: _f2py_c_args,
 )
 
+f2py = find_program('f2py')
+# It should be quite rare for the `f2py` executable to not be the one from
+# `numpy` installed in the Python env we are building for (unless we are
+# cross-compiling). If it is from a different env, that is still fine as long
+# as it's not too old. We are only using f2py as a code generator, and the
+# output is not dependent on platform or Python version (see gh-20612 for more
+# details).
+# This should be robust enough. If not, we can make this more complex, using
+# a fallback to `python -m f2py` rather than erroring out.
+f2py_version = run_command([f2py, '-v'], check: true).stdout().strip()
+if f2py_version.version_compare('<'+min_numpy_version)
+  error(f'Found f2py executable is too old: @f2py_version@')
+endif
+
+# Note: cannot handle .pyf.src or targets with #include's (due to no
+#       `depend_files` - see feature request at meson#8295)
```

Comment:
One more thing I just caught, `arpack.pyf.src` does not require `depend_files`. Could it be renamed to `arpack.pyf` and use `f2py_gen` as well?

Reply:
I will double check tomorrow, but IIRC those were two separate points (I'll tweak the comment to clarify):
1. cannot handle `.pyf.src`
2. cannot handle targets with `#include`'s (due to no `depend_files` - see feature request at meson#8295)

(1) is due to the naming: a generator needs to have fixed input/output name mappings, and `'_@BASENAME@module.c'` doesn't work here because only `.src` gets stripped rather than all of `.pyf.src`.

> Could it be renamed to `arpack.pyf`

It's a templated file rather than a regular signature file (everything in `<..>` is a template variable), so this won't work without special-casing `arpack.pyf` inside `generate_f2pymod` to indicate that template expansion is needed first. That isn't worth it.